interpreter/cling/lib/Utils/PlatformPosix.cpp

```diff
@@ -67,13 +67,15 @@ namespace {
           || P == lines[4] || P == lines[5] || P == lines[6] || P == lines[7])
         return true;
 
-      // There is a POSIX way of finding whether an address
-      // can be accessed for reading.
-      void *base = (void *)((((const size_t)P) / page_size) * page_size);
-      if (::msync(base, page_size, 0) != 0) {
-        assert(errno == EFAULT && "unexpected write error at address");
+      // Address of page containing P, assuming page_size is a power of 2
```

Comment:
Should we add an assert verifying the assumption?

Reply:
Sure, this is usually a property of the hardware, and all known platforms use power of 2. It makes no sense to use something else, since the caches would suffer if pages were not a power of 2.

Reply:
I created PR #4165 for this. Will merge it after merging #4162 and rebasing on top.